burotel/indico_burotel/cli.py

```diff
@@ -0,0 +1,151 @@
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import click
+import csv
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy.protection import ProtectionMode
+from indico.cli.core import cli_group
+from indico.modules.groups import GroupProxy
+from indico.modules.rb.models.locations import Location
+from indico.modules.rb.models.rooms import Room
+from indico.modules.users.models.users import User
+from indico.util.console import cformat
+
+
+ROOM_FIELDS = ('id', 'division', 'building', 'floor', 'number', 'verbose_name', 'owner', 'acl_entries')
+
+
+@cli_group(name='burotel')
+def cli():
+    """Manage the Burotel plugin."""
+
+
+def check_changed_fields(original, new):
+    diff = []
+    for field in ROOM_FIELDS:
+        if field == 'acl_entries':
+            orig_value = {e.principal for e in original.acl_entries}
+            target_value = new['acl_entries']
+        else:
+            orig_value = getattr(original, field)
+            target_value = new[field]
+        if orig_value != target_value:
+            diff.append((field, orig_value, target_value))
+    return diff
+
+
+def get_location(building):
+    location = Location.query.filter(Location.name == "Area {}".format(building)).first()
+    if not location:
+        location = Location(name="Area {}".format(building))
+        print(cformat("%{green!}+%{reset} Adding new location for building {}").format(building))
+        db.session.add(location)
+        db.session.flush()
+    return location
+
+
+def get_owner_from_email(email):
+    return User.query.filter(User.email == email).one()
+
+
+def change_room(room, changes):
+    for field, _, new_value in changes:
+        if field == 'acl_entries':
+            # clear the ACL and add the new principals
+            room.acl_entries.clear()
+            db.session.flush()
+            for p in new_value:
+                room.update_principal(p, full_access=True)
+        else:
+            setattr(room, field, new_value)
+
+
+def _print_changes(room, changes):
+    print '[{}]:'.format(room)
+    for field, old, new in changes:
+        if field == 'acl_entries':
+            old = {e.name for e in old}
+            new = {e.name for e in new}
+        print (cformat(' %{yellow}>%{reset} %{cyan}{}%{reset}: %{red}{}%{reset} -> %{green}{}%{reset}').
+               format(field, old, new))
```

Comment:
wrap before the `.`, currently it looks more like another argument to a function call